### PR TITLE
Remove image feature flag and fix original images

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | See below
| Type?             | refacto
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | Test that images generate normally.
| UI Tests          | 🟢 https://github.com/Hlavtox/ga.tests.ui.pr/actions/runs/7022274354
| Fixed issue or discussion?     | https://github.com/PrestaShop/PrestaShop/issues/32408, https://github.com/PrestaShop/PrestaShop/issues/32527, https://github.com/PrestaShop/PrestaShop/issues/34705, https://github.com/PrestaShop/PrestaShop/issues/32265
| Related PRs       | Autoupgrade PR must come that removes the feature flag from database and changes `PS_IMAGE_QUALITY` to `jpg` if it's set to `webp` or `webp_all`.
| Sponsor company   | 

### Description
- Remove all things related to multiple image format feature flag.
- Fixes original images uploaded for ALL OBJECTS, now they are kept in JPG, JPG/PNG or PNG format, depending on your configuration.
- Enables previously disabled test in format provider.
- Enables and fixes UI tests to test saving original image of products, stores and categories.
- Simplifies a way of rendering thumbnails, no need to pass $forceFormat now.
- All objects except categories are now resolved. They are the last thing to be fixed as last thing - https://github.com/PrestaShop/PrestaShop/issues/30520.

### Verified behavior for original images
- If you choose `Use PNG, if original image supports transparency`:
  - Upload JPG image (no transparency support), it gets saved as JPG format, `.jpg` thumbnails are JPG format.
  - Upload PNG image, it gets saved as PNG format, `.jpg` thumbnails are PNG format.
  - Upload WEBP image, it gets saved as PNG format, `.jpg` thumbnails are PNG format.
  - Upload GIF image, it gets saved as PNG format, `.jpg` thumbnails are PNG format.
  - etc.
- If you choose `Use JPG`:
  - Upload ANY image, it gets saved as JPG format, `.jpg` thumbnails are JPG format.
- If you choose `Use PNG`:
  - Upload ANY image, it gets saved as PNG format, `.jpg` thumbnails are PNG format.
- Tested for products, manufacturers, stores and categories. 
- `.webp` and `.avif` thumbnails are working correctly.

### Test files - JPG, PNG, WEBP
[images.zip](https://github.com/PrestaShop/PrestaShop/files/13487010/images.zip)

### Issues fixed
- https://github.com/PrestaShop/PrestaShop/issues/32408
- https://github.com/PrestaShop/PrestaShop/issues/32527
- https://github.com/PrestaShop/PrestaShop/issues/32265
- https://github.com/PrestaShop/PrestaShop/issues/34705

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]
It removes so much duplicated code 👍

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
we need approval from @PrestaShop/ui-tests-maintainers

**Comment by nesrineabdmouleh** [comment]
Ui tests => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/6421606776

**Review by boherm — DISMISSED** [review]
Cool! Thanks @Hlavtox!

**Review by nesrineabdmouleh — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]
Hi @Hlavtox 

just blocking this one temporarily not because of the content (all good for me) but because I don't understand something, I thought the feature was not finished and was supposed to be finished before we remove the feature flag.

So waiting for @PrestaShop/product-council opinion on this matter to know if/when the finalization of the feature is planned. Or if it's considered good enough to be a native feature as is

**Comment by Hlavtox** [comment]
@jolelievre There is nothing more to be done on this feature, it's finished.

There is some things to resolve, but it's not connected to multiple image format feature. https://github.com/PrestaShop/PrestaShop/issues/30520

**Comment by kpodemski** [comment]
cc @jolelievre

**Review by kpodemski — DISMISSED** [review]
There are some conflicts, but before resolving them, I suggest aligning with @jolelievre and @MatShir, so that there's a green light to remove the feature flag :)

**Comment by Hlavtox** [comment]
@MatShir @jolelievre @kpodemski Guys what's the final decision on this? 

And why should I wait for decision of someone from your company anyway?
I was the one who worked on this feature the most lately, don't I know the best if it's stable or not? :-)

**Comment by Hlavtox** [comment]
@PrestaShop/product-council @PrestaShop/committers I completely fixed all issues with the format of original uploaded images and improved the behavior. It was buggy because the webp implementation introduced in v8, not our flag. It fixes https://github.com/PrestaShop/PrestaShop/issues/32408, https://github.com/PrestaShop/PrestaShop/issues/32527, https://github.com/PrestaShop/PrestaShop/issues/34705.

@PrestaShop/qa-automation I enabled some previously disabled tests. You can finish the other ones when this is merged.

**Comment by Hlavtox** [comment]
@PrestaShop/qa-automation Guys could you please check&fix the ui tests? I tried but there is some error, it will be easy for you. :-) Edit my branch directly.

EDIT - CI Green, lets see what about full tests
EDIT - Now the full test suite should be green - https://github.com/Hlavtox/ga.tests.ui.pr/actions/runs/7022274354

**Comment by Hlavtox** [comment]
@matks @MatShir All issues resolved, tests green.

**Review by MatShir — DISMISSED** [review]
Thanks for your work Daniel. The last step to make the feature official is to fix the following bugs. So big green light from the @PrestaShop/product-council 💯 

Seen with @Progi1984, the success of the tests are enough to validate the bugs fixes. A brief QA manual pass could be performed to check everything is ok.

I let the QA automation @Progi1984 proceed to see if it all good.

File: tests/UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts:141]: It should be `webp` and not `png`.

File: tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts:140]:
```suggestion
      extImageType: 'webp',
```

It should be `webp` and not `png`.

**Review by Progi1984 — COMMENTED** [review]
Two small feedbacks

File: tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts
Comment (Hlavtox) [reply tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts:140]: No, it should be png ;) Only jpg or png for the original image.

File: tests/UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts
Comment (Hlavtox) [reply tests/UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts:141]: Same as the other one, we check the format inside. Always jpg or png, depending on configuration.

**Comment by Hlavtox** [comment]
@Progi1984 Answered, I check the format inside the file, not the extension. ;-)

Last step will be to enable the commented out webp category thumbnail test you guys prepared, but that will be done in another PR for categories only.

**Comment by Hlavtox** [comment]
Ping @PrestaShop/committers for approval, I want to work on this

**Comment by Progi1984** [comment]
> @Progi1984 Answered, I check the format inside the file, not the extension. ;-)
> 
> Last step will be to enable the commented out webp category thumbnail test you guys prepared, but that will be done in another PR for categories only.

I must check if it is ok with specs. I answer you in the week.

**Comment by Hlavtox** [comment]
@Progi1984 It is, like it was before it was messed up with 8.0.x webp implementation.

**Review by kpodemski — DISMISSED** [review]
Ok, so it looks good from the code perspective, but regarding the "How to test", I think it requires a full test of everything that is related to the images?

1. adding images to a new product
2. duplicating a new product
3. deleting a product (to see if images are correctly deleted)
and there's probably more

am I right?

File: tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts:140]: No, the original file must keep in the original format and must not be changed.

File: tests/UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts:141]: No, the original file must keep in the original format and must not be changed.

Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts:141]: The configuration impact generated files.

File: tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts:140]: The configuration impact generated files.

**Review by Progi1984 — CHANGES_REQUESTED** [review]
In UI Tests, original files are modified (and their original file types).

File: tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts
Comment (Hlavtox) [reply tests/UI/campaigns/functional/BO/08_design/06_imageSettings/13_checkProductImageFormat.ts:140]: @Progi1984 This is not true

**Comment by Hlavtox** [comment]
Sorry @Progi1984 but that is not true. I just tested it.

1. Install 1.7.8.
2. Set image configuration to "Use PNG, if the original image is PNG".
3. Add [jpgsource](https://github.com/PrestaShop/PrestaShop/assets/6097524/f2b89417-ff12-49be-89a5-478cf632fe3f) to product and check [/img/p/.../IMAGEID.jpg](https://github.com/PrestaShop/PrestaShop/assets/6097524/6d12ef23-12e2-42e4-905b-8ed5a5eae88d) ➡️ image is `JPG` with `.jpg` extension.
4. Add [pngsource](https://github.com/PrestaShop/PrestaShop/assets/6097524/3503b787-2d50-44a4-8aac-48db34e623ca) to product and check [/img/p/.../IMAGEID.jpg](https://github.com/PrestaShop/PrestaShop/assets/6097524/1fea2916-18d3-4ac0-ab90-676b7cd7ffb1) ➡️ image is `PNG` with `.jpg` extension.

**Review by ShaiMagal — APPROVED** [review]
@Progi1984 Daniel is right, it was always converted to the image format selected. 👍

**Comment by kpodemski** [comment]
@Progi1984 could you answer @Hlavtox and @ShaiMagal ?

**Comment by kpodemski** [comment]
ping @Progi1984

**Review by Progi1984 — APPROVED** [review]
OK for UI tests

**Review by kpodemski — APPROVED** [review]

**Review by M0rgan01 — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @Hlavtox 

Thank you for your PR, I tested it and it seems to works as we can see here : 

Imae on PNG for the store, reported by this issue #32527  :heavy_check_mark: : 

[recording(53).webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/804c05ab-298d-48a9-b830-dc07397151f2)

transparency on category asked on the same link, at the end :heavy_check_mark: 
image

> If you choose Use PNG, if original image supports transparency:
Upload JPG image (no transparency support), it gets saved as JPG format, .jpg thumbnails are JPG format.
Upload PNG image, it gets saved as PNG format, .jpg thumbnails are PNG format.
Upload WEBP image, it gets saved as PNG format, .jpg thumbnails are PNG format.
Upload GIF image, it gets saved as PNG format, .jpg thumbnails are PNG format.
etc.
If you choose Use JPG:
Upload ANY image, it gets saved as JPG format, .jpg thumbnails are JPG format.
If you choose Use PNG:
Upload ANY image, it gets saved as PNG format, .jpg thumbnails are PNG format.

it works as expected  :heavy_check_mark: 


Except : 

- I see that, as Proggy reported it on #32408, The 11_thumb.jpg is a webp originaly and at the end is still a PNG. 

[recording(51).webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/e74b8ba7-9c15-4024-9782-80fb930b80a3)


- I see that, as Proggy reported it on #32265, The 33.jpg is a webp originaly and at the end is still a PNG. 

[recording(52).webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/5e771091-bc98-40c0-9c65-3e65a50b7361)

for this two, @PrestaShop/product-council can you tell me if you have to keep the original format for the inital image ? 


For this point : 
>Tested for products, manufacturers, stores and categories.

When I create a new category, I don't have the webp create at the same time as you can see : 

[recording(54).webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/7906b0b8-cd76-45b1-8d2f-9d4a81e5dcf0)

It's only with category (for the store for example, we have directly the webp created). Do you correct this thing in this PR or you'll create another one for this issue ? 

Link to the auto test : https://github.com/AureRita/testing_pr/actions/runs/7490293502

Need the point of view of @PrestaShop/product-council for the point about unresolved points on the original format and yours for the last one. 

Except this. it's a QA :heavy_check_mark: because of all this test and the autotest :green_circle: 

Thank you

**Comment by MatShir** [comment]
> Except : 
> 
> - I see that, as Proggy reported it on #32408, The 11_thumb.jpg is a webp originaly and at the end is still a PNG. 
> 
> [recording(51).webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/e74b8ba7-9c15-4024-9782-80fb930b80a3)
> 
> 
> - I see that, as Proggy reported it on #32265, The 33.jpg is a webp originaly and at the end is still a PNG. 
> 
> [recording(52).webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/5e771091-bc98-40c0-9c65-3e65a50b7361)
> 
> for this two, @PrestaShop/product-council can you tell me if you have to keep the original format for the inital image ? 

For this part, it worked as expected. The orginal file is stored as jpg or png.

**Comment by MatShir** [comment]
Weird for the category, wdyt @Hlavtox ?

**Comment by Hlavtox** [comment]
@AureRita @MatShir As written several times on several places and in the PR description:

`All objects except categories are now resolved. They are the last thing to be fixed as last thing - Categories image - harmonization of 2 images  #30520.`

**Comment by ga-devfront** [comment]
Feature flag is removed from this PR => https://github.com/PrestaShop/autoupgrade/pull/681/files

**Pull request merged**
